src/actdoor.cpp

```diff
@@ -280,3 +281,29 @@ void actDoorFrame(Entity* my)
 		my->flags[PASSABLE] = true; // the actual frame should ALWAYS be passable
 	}
 }
+
+void Entity::doorHandleDamageMagic(int damage, Entity &magicProjectile, Entity *caster)
```

Comment:
Perhaps damage should be calculated within this function. So a fire ball spell will deal more damage than magic missile, all calculated here internally.